ui/content.go

```diff
@@ -22,7 +25,15 @@ type content struct {
 	sources []string
 
 	// This is only created in prd-mode, the pre-parsed template
-	tpl *template.Template
+	// HTML template - used for serving HTML content
+	htpl *htemplate.Template
+
+	// This is only created in prd-mode, the pre-parsed template
+	// Text template - currently used for /open_search.xml only
+	ttpl *ttemplate.Template
```

Comment:
Seems like you should be able to better take advantage of the fact that `html/template.Template.Execute` and `text/template.Template.Execute` are structurally equivalent. You should be able to store both types into an interface field.
```
type renderer interface {
    Execute(w io.Writer, data interface{}) error
}
```
That would at least clean up one set of switch statements for dispatching.

Reply:
Thanks @kellegous, I was thinking of using an interface in this case, but wasn't able to figure out how to use one. I'll try again and push the fixes.